packages/rum-core/src/domain/rumEventsCollection/view/viewMetrics/trackLargestContentfulPaint.spec.ts

```diff
@@ -1,30 +1,53 @@
 import type { RelativeTime } from '@datadog/browser-core'
-import { DOM_EVENT, noop } from '@datadog/browser-core'
+import {
+  DOM_EVENT,
+  ExperimentalFeature,
+  addExperimentalFeatures,
+  noop,
+  resetExperimentalFeatures,
+} from '@datadog/browser-core'
 import { restorePageVisibility, setPageVisibility, createNewEvent } from '@datadog/browser-core/test'
 import type { RumLargestContentfulPaintTiming } from '../../../../browser/performanceCollection'
 import type { TestSetupBuilder } from '../../../../../test'
 import { setup } from '../../../../../test'
+import type { LifeCycle } from '../../../lifeCycle'
 import { LifeCycleEventType } from '../../../lifeCycle'
 import type { RumConfiguration } from '../../../configuration'
 import { resetFirstHidden } from './trackFirstHidden'
 import { LCP_MAXIMUM_DELAY, trackLargestContentfulPaint } from './trackLargestContentfulPaint'
 
 describe('trackLargestContentfulPaint', () => {
   let setupBuilder: TestSetupBuilder
-  let lcpCallback: jasmine.Spy<(value: RelativeTime) => void>
-  let eventTarget: Window
+  let lcpCallback: jasmine.Spy<(value: RelativeTime, targetSelector?: string) => void>
   let configuration: RumConfiguration
+  let target: HTMLImageElement
+
+  function newLargestContentfulPaint(lifeCycle: LifeCycle, overrides?: Partial<RumLargestContentfulPaintTiming>) {
+    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, [
+      {
+        entryType: 'largest-contentful-paint',
+        startTime: 789 as RelativeTime,
+        size: 10,
+        element: target,
+        ...overrides,
+      },
+    ])
+  }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏how about extracting some kind of performance entry builder in fixtures.ts, like what we have for events?

```ts
createPerformanceEntry<T>(entryType, overrides) {
  switch (entryType) {
    ...
  } 
}
```

Reply:
Good idea! I created it. For now, I only used the fixures for the Core Web Vital but I'll do a followup PR to extend its usage to all the tests.